### PR TITLE
Support dependency outputs

### LINKED ISSUE #432: Wire up a bundle's outputs into bundle.dependencies.NAME.outputs
Once we have #344 implemented and #430, then we can collect output from bundle dependencies, and inject them into the bundle as `bundle.dependencies.NAME.outputs`

### PR DESCRIPTION
**Pull request by carolynvs**
* Inject the output values for each dependency into `/cnab/app/dependencies/ALIAS/outputs`
* Inject the bundle.json for each dependency into `/cnab/app/dependencies/ALIAS/bundle.json`
* Inject `bundle.dependencies.ALIAS.outputs.OUTPUT`

Closes #432

### CODE REVIEW

File: build/testdata/bundles/wordpress/porter.yaml [hunk 1]
```diff
@@ -43,7 +43,7 @@ install:
         #externalDatabase.database: "{{ bundle.dependencies.mysql.parameters.database-name }}"
```
Comment (carolynvs) [inline build/testdata/bundles/wordpress/porter.yaml]: I have another issue to address `bundle.dependencies.DEP.parameters.PARAM` #533

File: pkg/cnab/provider/uninstall.go [hunk 2]
```diff
@@ -42,12 +41,13 @@ func (d *Duffle) Uninstall(args ActionArguments) error {
...
 
-	driver, err := d.newDriver(args.Driver, c.Name)
+	driver, err := d.newDriver(args.Driver, c.Name, args)
 	if err != nil {
 		return errors.Wrap(err, "unable to instantiate driver")
 	}
 	i := action.Uninstall{
-		Driver: driver,
+		Driver:          driver,
+		OperationConfig: args.ApplyFiles(),
```
Comment (carolynvs) [inline pkg/cnab/provider/uninstall.go]: This change here is part of a PR to cnab-go (see https://github.com/deislabs/cnab-go/pull/112). In the meantime, we'll use my fork.

File: pkg/config/runtime-manifest.go [hunk 3]
```diff
@@ -147,43 +189,85 @@ func (m *RuntimeManifest) buildSourceData() (map[string]interface{}, error) {
...
+
+		// bundle.dependencies.ALIAS.outputs.NAME
+		depBundle := make(map[string]interface{})
+		deps[alias] = depBundle
+		depOutputs := make(map[string]interface{})
+		depBundle["outputs"] = depOutputs
+
+		if bun.Outputs == nil || m.Action == ActionUninstall {
+			// uninstalls are done backwards, so we don't have outputs available from dependencies
+			// TODO: validate that they weren't trying to use them at build time so they don't find out at uninstall time.
```
Comment (carolynvs) [inline pkg/config/runtime-manifest.go]: See https://github.com/deislabs/porter/issues/532

File: pkg/config/runtime-manifest.go [hunk 4]
```diff
@@ -147,43 +189,85 @@ func (m *RuntimeManifest) buildSourceData() (map[string]interface{}, error) {
...
+
 	deps := make(map[string]interface{})
-	bundle["dependencies"] = deps
-	for name, dependency := range m.Dependencies {
-		dep, sensitives, err := dependency.resolve()
-		if err != nil {
-			return nil, err
+	bun["dependencies"] = deps
+	for alias, bun := range m.bundles {
+		// TODO: Do we still want to allow resolving against bundle.dependencies.ALIAS.parameters.NAME and credentials?
```
Comment (carolynvs) [inline pkg/config/runtime-manifest.go]: We aren't going to support bundle.dependencies.DEP.credentials anymore. I will come back in another PR and remove that from the docs and examples.

File: Makefile [hunk 5]
```diff
@@ -79,7 +79,7 @@ endif
 
 test: clean-last-testrun test-unit test-integration test-cli
 
-test-unit: build
+test-unit: build-porter
```
Comment (vdice) [inline Makefile]: Yay!

File: pkg/config/outputs.go [hunk 6]
```diff
@@ -49,15 +49,15 @@ func (c *Config) ReadBundleOutput(name string, claim string) (*Output, error) {
 
 	outputPath := filepath.Join(bundleOutputsDir, name)
 
-	bytes, err := c.Context.FileSystem.ReadFile(outputPath)
+	bytes, err := c.FileSystem.ReadFile(outputPath)
 	if err != nil {
-		return nil, errors.Errorf("unable to read output '%s' for claim '%s'", name, claim)
+		return nil, errors.Errorf("unable to read output %q for claim %q", name, claim)
```
Comment (vdice) [inline pkg/config/outputs.go]: nice; `%q` = new learning for me! handy.

File: tests/dependencies_test.go [hunk 7]
```diff
@@ -24,7 +24,9 @@ func TestDependenciesLifecycle(t *testing.T) {
 
 	upgradeWordpressBundle(p, namespace)
 
-	invokeWordpressBundle(p, namespace)
+	upgradeWordpressBundle(p, namespace)
```
Comment (vdice) [inline tests/dependencies_test.go]: Should we remove this extra `upgradeWordpressBundle(p, namespace)` invocation?

Comment (vdice) [reply tests/dependencies_test.go]: Should we create a TODO to uncomment `invokeWordpressBundle(p, namespace)` ?  And/or add a comment mentioning what is currently preventing a successful test?

Comment (carolynvs) [reply tests/dependencies_test.go]: Oops! Thanks, I commented that out when I was debugging something and forgot to put it back in. Fixing....

Comment (carolynvs) [reply tests/dependencies_test.go]: Same for the extra upgrade call, I just removed that and pushed up the fix.

**Comment by carolynvs** [comment]
Rebased

**Pull request merged**